**Importing the dataset**

In [1]:
from tensorflow.keras.datasets import mnist
(train_images,train_labels),(test_images,test_labels) = mnist.load_data()


**Building the network**

In [2]:
from tensorflow import keras
from tensorflow.keras import layers
model = keras.Sequential([
    layers.Dense(512,activation = "relu"),
    layers.Dense(10, activation = "softmax")    
])

In [3]:
model.compile(
    optimizer = "rmsprop",
    loss = "sparse_categorical_crossentropy",
    metrics = ["accuracy"]
)

**Reshaping the image data and changing its data type and values in [0,1]**

In [4]:
train_images = train_images.reshape((60000, 28*28))
train_imagese = train_images.astype("float32")/255

test_images = test_images.reshape((10000, 28*28))
test_imagese = test_images.astype("float32")/255

**Fitting the model**

In [5]:
model.fit(train_images, train_labels, epochs = 5, batch_size = 128)

Epoch 1/5
469/469 [==============================] - 4s 7ms/step - loss: 5.5934 - accuracy: 0.8998
Epoch 2/5
469/469 [==============================] - 4s 8ms/step - loss: 0.6513 - accuracy: 0.9529
Epoch 3/5
469/469 [==============================] - 3s 7ms/step - loss: 0.4634 - accuracy: 0.9645
Epoch 4/5
469/469 [==============================] - 4s 7ms/step - loss: 0.3622 - accuracy: 0.9722
Epoch 5/5
469/469 [==============================] - 3s 7ms/step - loss: 0.3216 - accuracy: 0.9755


In [6]:
test_digits = test_images[0:10]

**Model prediction**

In [7]:
pred = model.predict(test_digits)

1/1 [==============================] - 0s 68ms/step


In [8]:
pred[0]

array([0., 0., 0., 0., 0., 0., 0., 1., 0., 0.], dtype=float32)

In [9]:
pred[1]

array([0.0000000e+00, 3.1046998e-27, 1.0000000e+00, 5.7269918e-36,
       0.0000000e+00, 0.0000000e+00, 0.0000000e+00, 0.0000000e+00,
       0.0000000e+00, 0.0000000e+00], dtype=float32)

In [10]:
pred[8]

array([0.0000000e+00, 0.0000000e+00, 0.0000000e+00, 0.0000000e+00,
       7.7858277e-19, 1.0000000e+00, 1.8804917e-25, 0.0000000e+00,
       2.7908603e-19, 5.4742579e-25], dtype=float32)

In [11]:
#Returning the index with highest probabilty
pred[8].argmax()

5

_This means that the image is most probably of index 6, i.e the digit 6_ 

**Evaluating accuracy**

In [12]:
test_loss,test_acc = model.evaluate(test_images,test_labels)
print("Test accuracy: ", test_acc)

313/313 [==============================] - 1s 3ms/step - loss: 0.7114 - accuracy: 0.9593
Test accuracy:  0.9592999815940857


_As we can see, the test accuracy (95.6%) is less than train_accuracy (97.5%)_